tss-esapi/tests/integration_tests/attributes_tests/session_attributes_tests.rs

```diff
@@ -0,0 +1,149 @@
+        1u8.checked_shl(6)
+            .expect("Failed to create 'encrypt' value"),
+        encrypt
+    );
+    test_valid_conversion!(
+        1u8.checked_shl(7).expect("Failed to create 'audit' value"),
+        audit
+    );
+}
+
+#[test]
+fn test_invalid_session_attributes_conversions() {
+    // bit 3 and 4 are reserved and shall be cleared.
+    // No error for this is implemented.
+    // See https://github.com/parallaxsecond/rust-tss-esapi/issues/330
+}
+
+#[test]
+fn test_valid_session_attributes_mask_conversions() {
+    test_valid_mask_conversion!(
+        1u8.checked_shl(0)
+            .expect("Failed to create 'use continue session' mask value"),
+        "use_continue_session"
+    );
+    test_valid_mask_conversion!(
+        1u8.checked_shl(1)
+            .expect("Failed to create 'use audit exclusive' mask value"),
+        "use_audit_exclusive"
+    );
+    test_valid_mask_conversion!(
+        1u8.checked_shl(2)
+            .expect("Failed to create 'use audit reset' mask value"),
+        "use_audit_reset"
+    );
+    test_valid_mask_conversion!(
+        1u8.checked_shl(5)
+            .expect("Failed to create 'use decrypt' mask value"),
+        "use_decrypt"
+    );
+    test_valid_mask_conversion!(
+        1u8.checked_shl(6)
+            .expect("Failed to 'use encrypt' mask value"),
+        "use encrypt"
+    );
+    test_valid_mask_conversion!(
+        1u8.checked_shl(7)
+            .expect("Failed to create 'use audit' session value"),
+        "use audit"
+    );
+}
+
+#[test]
+fn test_invalid_session_attributes_mask_conversions() {
+    // bit 3 and 4 are reserved and shall be cleared.
+    // No error for this is implemented.
+    // See https://github.com/parallaxsecond/rust-tss-esapi/issues/330
+}
+
+#[test]
+fn test_session_attributes_builder_constructing() {
+    let _b1 = SessionAttributes::builder();
+    let _b2 = SessionAttributesMask::builder();
+    let _b3 = SessionAttributesBuilder::default();
+    let _b4 = SessionAttributesBuilder::new();
+}
+
+#[test]
+fn test_session_attributes_builder() {
+    let expected_session_attributes = SessionAttributes::from(0b11100111u8);
+    let expected_session_attributes_mask = SessionAttributesMask::from(0b11100111u8);
+
+    let (actual_session_attributes, actual_session_attributes_mask) =
+        SessionAttributesBuilder::new()
+            .with_continue_session(true)
+            .with_audit_exclusive(true)
+            .with_audit_reset(true)
+            .with_decrypt(true)
+            .with_encrypt(true)
+            .with_audit(true)
+            .build();
+
+    assert_eq!(
+        expected_session_attributes, actual_session_attributes,
+        "SessionAttributes builder did not produce the expected session attributes value"
+    );
+    assert_eq!(
+        expected_session_attributes_mask, actual_session_attributes_mask,
+        "SessionAttributes builder did not produce the expected session attributes mask value"
+    )
+}
```

Comment:
Could you add one like this, but with `0u8` as the value and default builder output?

Reply:
Done

Reply:
Don't think you've pushed this